tensorboard/BUILD

```diff
@@ -96,6 +96,24 @@ py_test(
     ],
 )
 
+py_test(
+    name = "manager_e2e_test",
+    size = "large",  # spawns subprocesses, sleeps, makes requests to localhost
+    timeout = "short",  # about 15 seconds on my machine
+    flaky = True,  # on Python 2, fails about 0.5% of the time
+    srcs = ["manager_e2e_test.py"],
+    srcs_version = "PY2AND3",
+    visibility = ["//tensorboard:internal"],
+    deps = [
+        ":manager",
+        "//tensorboard:expect_tensorflow_installed",
+        "@org_pythonhosted_six",
+    ],
+    data = [
+        ":tensorboard",
```

Comment:
Is this required? Is it because manager does not declare dependency on `:tensorboard` or something?

Reply:
Yes, it’s required. This isn’t a Python dependency; it’s a dependency on
the built `tensorboard(1)` binary, which we execute as a subprocess.

Reply:
I was assuming that it was indirectly part of `:manager`'s `data`. Don't see the test directly invoking the binary so I thought it was a bit weird.

Reply:
`:manager` invokes whatever `tensorboard(1)` is on the system path.
`:manager` can’t depend on `:tensorboard`, because `:tensorboard`
depends on it! But the test can depend on `:tensorboard` and add that to
the system path so that `:manager` picks up the binary:

```python
    # Add our Bazel-provided `tensorboard` to the system path.
    tensorboard_binary_dir = os.path.realpath("./tensorboard/")
    path_environ = {
        "PATH": os.pathsep.join((tensorboard_binary_dir, os.environ["PATH"])),
    }
    path_environ_patcher = mock.patch.dict(os.environ, path_environ)
    path_environ_patcher.start()
```

It doesn’t directly _invoke_ it, but it does directly _consume_ it (by
adding it to the path as listed above).